### PR TITLE
edit original post

### PR DESCRIPTION
**Pull request by tbg**
hopefully making it better.

[Review on Reviewable](https://reviewable.io/reviews/tschottdorf/goplay/2)

### CODE REVIEW

File: cgo_libc_static/README.md [hunk 1]
```diff
@@ -22,87 +31,77 @@ func main() {
 }
 ```
 
-Looks about as innocuous as nonsensical, right? If we run it naively, nothing happens:
+Looks about as innocuous as nonsensical, right? If we run it naively, nothing
```
Comment (petermattis) [inline cgo_libc_static/README.md]: `s/as nonsensical/as it is nonsensical/g`

File: cgo_libc_static/README.md [hunk 2]
```diff
@@ -22,87 +31,77 @@ func main() {
...
+## Discovery
 
-Proud mother of the little critter is [cockroachdb/cockroach#3310](https://github.com/cockroachdb/cockroach/pull/3310). Basically, [@tamird](https://github.com/tamird) was building a static test binary with the goal of running it during nightly builds. The test uses [lib/pq](https://github.com/lib/pq) to connect to a [Cockroach DB cluster](http://www.cockroachlabs.com) (which essentially speaks Postgres' SQL dialect). You already know what happened when he tried to run it:
+This bug hit us out of the blue in
+[cockroachdb/cockroach#3310](https://github.com/cockroachdb/cockroach/pull/3310).
+Basically, [@tamird](https://github.com/tamird) was building a static test
+binary with the goal of running it during nightly builds. The test uses
+[lib/pq](https://github.com/lib/pq) to connect to a [Cockroach DB
+cluster](http://github.com/cockroachdb/cockroach) (which essentially speaks
+Postgres' SQL dialect). You already know what happened when he tried to run it:
```
Comment (petermattis) [inline cgo_libc_static/README.md]: `s/Postgres' SQL dialect/Postgres' wire protocol/g`

File: cgo_libc_static/README.md [hunk 3]
```diff
@@ -152,21 +152,37 @@ func (cn *conn) ssl(o values) {
 }
 ```
 
-It's relatively easy to guess this in this heavily truncated version, but there's actually a successful call to `user.Current()` from `userCurrent()` (marked with `!!!`). We only saw this after adding an `fmt.Println()` in `user.Current()` and wondered why that printed more than we expected. So, that's weird - the crash is either random<sup>2</sup> or it depends on something else happening before it.
+It's relatively easy to guess this in this heavily truncated version, but
+there's actually a successful call to `user.Current()` from `userCurrent()`
+(marked with `!!!`). We only saw this after adding an `fmt.Println()` in
```
Comment (petermattis) [inline cgo_libc_static/README.md:159]: Should that be `a fmt`, or is `an fmt` correct?

File: cgo_libc_static/README.md [hunk 4]
```diff
@@ -244,17 +249,28 @@ and the following patch to `$(go env GOROOT)/src/net/fd_unix.go`:
 +        user.Current()
```
Comment (petermattis) [inline cgo_libc_static/README.md:252]: On line 236: `s/without this, don't/without this, we don't/g`?

**Comment by petermattis** [comment]
LGTM

File: cgo_libc_static/README.md (hunk 3 above)
Comment (tbg) [inline cgo_libc_static/README.md:159]: I'm "eff-em-tee", but apparently the majority is "fumt", so changed to `a`.

File: cgo_libc_static/README.md (hunk 2 above)
Comment (tbg) [inline cgo_libc_static/README.md]: done.

File: cgo_libc_static/README.md (hunk 1 above)
Comment (tbg) [inline cgo_libc_static/README.md]: done.

File: cgo_libc_static/README.md (hunk 4 above)
Comment (tbg) [inline cgo_libc_static/README.md:252]: done.

File: cgo_libc_static/README.md [hunk 5]
```diff
@@ -111,26 +110,27 @@ fatal error: unexpected signal during runtime execution
 goroutine 3890 [syscall, locked to thread]:
 runtime.cgocall(0x44c7f0, 0xc82036a8d8, 0xc800000000)
     /usr/local/go/src/runtime/cgocall.go:120 +0x11b fp=0xc82036a888 sp=0xc82036a858
-os/user._Cfunc_mygetpwuid_r(0x0, 0xc8203a8390, 0x7f3c5c000a10, 0x400, 0xc8200e4058, 0x7f3c00000000)
```
Comment (tamird) [inline cgo_libc_static/README.md:114]: I wouldn't remove this, this is the symbol you'd actually find in the symbol table

File: cgo_libc_static/README.md [hunk 6]
```diff
@@ -366,30 +388,54 @@ Aha! When `%fs = 99`, apparently all is well, but in an iteration which has
...
+statically against glibc, it's really nothing you should ever find yourself
+doing. At least not if you're using threads.
+
+# Conclusion
+
+Linking statically against `glibc` has proven to be an insane idea, but it's
+surprising that this was apparently news for everyone up to (but not including)
+the glibc bug tracker.
+
+We figured out that we can [get a less obviously ludacris static build](https://github.com/cockroachdb/cockroach/pull/3343)
```
Comment (tamird) [inline cgo_libc_static/README.md]: s/ludacris/ludicrous/

Comment (tbg) [inline cgo_libc_static/README.md]: TIL. Done.

File: cgo_libc_static/README.md (hunk 5 above)
Comment (tbg) [inline cgo_libc_static/README.md:114]: hmm, good point. Reinstating.

**Pull request merged**